### PR TITLE
Converted Feature Summary

### PR DESCRIPTION
**Pull request by ryan-rushton**
Updated a couple of components. Handled date formatting using i18next. Made it so that real update times are shown. We need to do the last updated by bit but it would be good to get that data from the backend or token or something like that.

@xfeier53, I think those changes will fix the styling we were looking at on Friday. There top border of the feature list is weird when it is collapsed but I can't see exactly where it is coming from.

@kevinoshea I wanted to use the response header date to set the last updated stuff but I need to modify the rest layer so I might leave that until you finish with it. That won't effect responsiveness so I figure it might be best to update it based off that.

### CODE REVIEW

File: frontend/src/models/Feature.ts [hunk 1]
```diff
@@ -25,6 +27,8 @@ export const createFeatureFromFetchedData = (data: any): Feature => {
     originalAutomatedStatus: data.originalAutomatedStatus,
     tags: data.tags,
     scenarios: data.elements.map(element => createScenarioFromFetchedData(element)),
+    lastEditedBy: data.statusLastEditedBy || undefined,
```
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply frontend/src/models/Feature.ts]: It comes through from the api as null, technically optional fields are something or undefined so I made sure it is set to that. I cant remember if TS forces nulls to undefined or spits the dummy but I think it is the later. Will investigate

File: frontend/src/modules/report/FeatureListContainer/ListViewFeatureList/ListViewFeatureList.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  selectedFeatureId: string;
+  selectedStatus: StatusMap<boolean>;
+  handleFeatureSelected(feature: Feature): void;
+}
+
+const ListViewFeatureList: FC<Props> = ({ featureList, selectedFeatureId, selectedStatus, handleFeatureSelected, classes }) => {
+  const filterFeatureList = featureList.filter(feature => selectedStatus[feature.calculatedStatus]);
+
+  const classesMap: StatusMap<string> = {
+    [Passed]: classes.itemPassed,
```
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply frontend/src/modules/report/FeatureListContainer/ListViewFeatureList/ListViewFeatureList.tsx]: No the classes thing is directly generated by material-ui. So classes.itemPassed is really something auto generated like this, `TagViewFeatureList-itemPassed-568`.

File: frontend/src/modules/report/FeatureSummary/FeatureSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    [Skipped]: classes.featureSkipped,
+  };
+
+  return (
+    <Card raised className={classes.featureSummary}>
+      <Grid container>
+        <Grid item xs={3}>
+          <StatusIcons firstStatus={feature.originalAutomatedStatus} secondStatus={feature.calculatedStatus} />
+          <div className={classes.featureEditInfo}>
+            {feature.lastEditedBy && <div>{t('report.lastEditedByOn', { name: feature.lastEditedBy, date: feature.lastEditedOn })}</div>}
```
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply frontend/src/modules/report/FeatureSummary/FeatureSummary.tsx]: Yeah there is the potential to see dates which aren't quite right then but I don't think that will actually cause issues. Might need to talk about concurrency again as I don't quite remember where we landed on that while Feier was working on this.

File: frontend/src/models/Feature.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply frontend/src/models/Feature.ts]: It doesn't seem to break anything but I don't know if I like it considering TS's stance that null and undefined are not compatible. I could change the lastEditedBy type to `string | null` but then we lose out on the nice optional syntax around the app when working with it. I can go either way though so let me know if you have a preference.

File: frontend/src/models/Status.ts [hunk 4]
```diff
@@ -5,4 +5,8 @@ enum Status {
   Undefined = 'undefined',
 }
 
+export type StatusMap<T> = { [key in Status]: T };
```
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply frontend/src/models/Status.ts]: Yeah I really like type aliases for this kinda thing :D

File: frontend/src/models/Feature.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinoshea) [reply frontend/src/models/Feature.ts]: Ok yep, thought that might be the case. Lets stick with what you've got for now, we can review again when moving these to a service call and doing proper type checks on the response.

File: frontend/src/modules/report/FeatureSummary/FeatureSummary.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevinoshea) [reply frontend/src/modules/report/FeatureSummary/FeatureSummary.tsx]: Ah yes concurrency... we should think about that.

**Pull request merged**